src/main/java/com/synopsys/integration/detect/lifecycle/boot/decision/AutoScanTypeDecider.java

```diff
@@ -0,0 +1,165 @@
+        if (!hasImageOrTar && detectConfiguration.getValue(DetectProperties.DETECT_AUTONOMOUS_SCAN_ENABLED)) {
+            Path detectSourcePath = detectConfiguration.getPathOrNull(DetectProperties.DETECT_SOURCE_PATH);
+            AllNoneEnumCollection<DetectTool> includedTools = detectConfiguration.getValue(DetectProperties.DETECT_TOOLS);
+            AllNoneEnumCollection<DetectTool> excludedTools = detectConfiguration.getValue(DetectProperties.DETECT_TOOLS_EXCLUDED);
+            if (detectSourcePath == null) {
+                logger.error("Detect autonomous scan mode requires Detect Source Path (--detect.source.path) to be set.");
+            } else {
+                // This map has individual paths to files grouped under types.
+                PathsCollection pathsCollection = search(detectSourcePath);
+                logger.info("includedTools: {}", includedTools.toPresentValues());
+                logger.info("excludedTools: {}", excludedTools.toPresentValues());
+                final Map<DetectTool, Set<String>> scanTypeEvidenceMap = new HashMap<>();
+                if (!excludedTools.containsValue(DetectTool.BINARY_SCAN)
+                        && !includedTools.containsValue(DetectTool.BINARY_SCAN)
+                        && !pathsCollection.binaryPaths.isEmpty()) {
+                    scanTypeEvidenceMap.put(DetectTool.BINARY_SCAN, pathsCollection.binaryPaths);
+                }
+                if (!excludedTools.containsValue(DetectTool.DETECTOR)
+                        && !includedTools.containsValue(DetectTool.DETECTOR)
+                        && !pathsCollection.detectorPaths.isEmpty()) {
+                    scanTypeEvidenceMap.put(DetectTool.DETECTOR, pathsCollection.detectorPaths);
+                }
+                if (!excludedTools.containsValue(DetectTool.SIGNATURE_SCAN)
+                        && !includedTools.containsValue(DetectTool.SIGNATURE_SCAN)
+                        && !pathsCollection.signaturePaths.isEmpty()) {
+                    scanTypeEvidenceMap.put(DetectTool.SIGNATURE_SCAN, pathsCollection.signaturePaths);
+                }
+                return scanTypeEvidenceMap;
+            }
+        }
+        return Collections.EMPTY_MAP;
+    }
+    
+    private final Set<String> avoidAbsolutley = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            ".gitattributes", 
+            ".gitignore", 
+            ".github", 
+            ".git",
+            "__MACOSX",
+            ".DS_Store")));
+    
+    /**
+     * This should be replaced with an intuitive binary reader that can zip and upload a stream of image formats
+     * since a malicious binary can be embedded in an image file. The original paths to the 
+     * individual image files should be stored in a text file for the BDBA worked to 
+     * reference in its report.
+     */
+    private final Set<String> ignoreReluctantly = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
```

Comment:
There will be no (in earnest) similarities to CLL names.
Static not justified here.

Reply:
1) CLL was a suggestion.  I referred to it because we do a similar file walk in that library. Something like `filesToAvoid` or `fileExtensionsToAvoid` are my suggestions to make the set names more meaningful with respect to the data they hold.
2) Interested to know why you think static is not justified here.